File: llm-server/pages/api/ingest.ts
import type {NextApiRequest, NextApiResponse} from 'next';
import pdfHandler from '@/data-sources/pdfHandler';
import websiteHandler from '@/data-sources/websiteHandler';
import codebaseHandler from "@/data-sources/codebaseHandler";

export default async function handler(req: NextApiRequest, res: NextApiResponse) {
    try {
        const {type} = req.body;

        if (req.method !== 'POST') {
            return res.status(405).json({error: 'Method not allowed'});
        }

        if (!type) {
            return res.status(400).json({message: 'No type in the request'});
        }

        const supportedTypes = ['pdf', 'website', 'codebase'];
        if (!supportedTypes.includes(type)) {
            return res.status(400).json({message: 'Type not supported'});
        }

        if (type === 'pdf') {
            return await pdfHandler(req, res);
        } else if (type === 'website') {
            return await websiteHandler(req, res);
        } else if (type === 'codebase') {
            return await codebaseHandler(req, res);
        } else {
            return res.status(400).json({message: 'Not supported type'});
        }

    } catch (e) {
        console.error(e);
        // Return error message and line number
        // @ts-ignore
        return res.status(500).json({error: e.message, line: e.lineNumber});
    }
}

File: dj_backend_server/api/views/ingest.py
from django.http import JsonResponse
from django.views.decorators.http import require_POST

from api.data_sources import codebase_handler, pdf_handler, website_handler
import json

@require_POST
def handler(request):
    try:
        data = json.loads(request.body.decode('utf-8'))
        type_ = data['type']

        if type_ not in ('pdf', 'website', 'codebase'):
            return JsonResponse({'error': 'Type not supported'})

        if type_ == 'pdf':
            return pdf_handler(request)
        elif type_ == 'website':
            return website_handler(request)
        elif type_ == 'codebase':
            return codebase_handler(request)

    except Exception as e:
        return JsonResponse({
            'error': str(e),
            'line': e.__traceback__.tb_lineno,
        })
